DirectProgramming/DPC++FPGA/Tutorials/Features/device_global/src/device_global.cpp

```diff
@@ -0,0 +1,112 @@
+//==============================================================
+// Copyright Intel Corporation
+//
+// SPDX-License-Identifier: MIT
+// =============================================================
+#include <math.h>
+
+#include <CL/sycl.hpp>
+#include <iostream>
+#include <sycl/ext/intel/fpga_extensions.hpp>
+
+// dpc_common.hpp can be found in the dev-utilities include folder.
+// e.g., $ONEAPI_ROOT/dev-utilities//include/dpc_common.hpp
+#include "dpc_common.hpp"
+
+using namespace sycl;
+using namespace sycl::ext::oneapi::experimental;
+
+constexpr size_t kNumCounters = 4;
+constexpr int kInitialValue = 10;
+constexpr unsigned kNumIncrements = 3;
+
+using IntScalar = std::array<int, 1>;
+using FPGAProperties = decltype(properties(device_image_scope));
+
+// Array of counters that have a lifetime longer than a single kernel invocation
+device_global<int[kNumCounters], FPGAProperties> counters;
+// Flag if the counters have been initialized
+device_global<bool, FPGAProperties> is_counters_initialized;
```

Comment:
C++ definition of zero-initialization guarantees this. And the spec for `device_globals` states that we are using zero-initialization when `device_globals` are default constructed.

Reply:
Will this translate to the generated HW? 
I think we had similar issues in the past, where the emulation worked as expected because it was following the C++ standard, but the HW that we generated ended up not being aligned

Reply:
We are actively working to force the HW to be aligned to this. It is not implemented yet. The release of this code sample should align with 2023.1, but our internal deadline for the sample is 2023.0